standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java

```diff
@@ -9103,6 +9107,14 @@ public static void main(String[] args) throws Throwable {
                 + e.getMessage(), e);
           }
         }
+        // Remove from zookeeper if it's configured
+        try {
+          if (zooKeeperHelper != null) {
```

Comment:
might be better to use the same configuration (as the one to register), to identify if zookeeper de-registration should happen and for some reason if that is null here then we should actually log an exception since thats is an unexpected state 

MetastoreConf.getVar(conf, ConfVars.THRIFT_SERVICE_DISCOVERY_MODE)
            .equalsIgnoreCase("zookeeper")

Reply:
That looks better. Done.